src/types/redis_string.cc

```diff
@@ -384,7 +384,7 @@ rocksdb::Status String::MSet(const std::vector<StringPair> &pairs, uint64_t ttl)
     batch->PutLogData(log_data.Encode());
     AppendNamespacePrefix(pair.key, &ns_key);
     batch->Put(metadata_cf_handle_, ns_key, bytes);
-    LockGuard guard(storage_->GetLockManager(), ns_key);
+    if (lock) LockGuard guard(storage_->GetLockManager(), ns_key);
```

Comment:
Would lifetime of LockGuard be a temperal?

```
std::optional<LockGuard> guard;
if (lock) {
  guard = ...
}
```

What about this?

Reply:
Yes, this lock wouldn't work since it will release the lock after exiting the if block. It also has a problem with the previous implementation which wrote the batch per key, we could change it to a single batch in this PR.

Reply:
> It also has a problem with the previous implementation which wrote the batch per key, we could change it to a single batch in this PR.

i want to put this in another PR, just to make the PRs clear (the diff should be small and easy to follow), WDYT?

Reply:
@enjoy-binbin It makes a lot of sense.